### PR TITLE
CAMS-767 Filter Trustee List by Appointment Status

### PR DESCRIPTION
**Pull request by fmaddenflx**
## Summary
- Add status filter (Active/Inactive) to trustee list with mobile-first responsive layout
- Improve filter controls with paired wrapping (4→2→1 columns)
- Replace hardcoded hex colors with SCSS variables across the UI
- Tune table column proportions to prevent horizontal overflow at all viewport widths

## Test plan
- [ ] Verify status filter correctly filters trustees by Active/Inactive
- [ ] Verify filter layout wraps in pairs at medium widths and stacks on mobile
- [ ] Verify table displays without horizontal scroll at all widths (880px+)
- [ ] Verify status column displays full text without wrapping
- [ ] Verify no visual regressions in other filter controls

## Summary by Sourcery

Add status-based filtering and responsive layout improvements to the trustees list while standardizing UI colors and table behaviors across the app.

New Features:
- Introduce an Active/Inactive/All status filter on the trustees list and expose it via a Status combobox in the filter panel.

Bug Fixes:
- Ensure trustees with mixed active and inactive appointments appear in both views and that status text in the list clearly distinguishes inactive sub-statuses.
- Prevent horizontal overflow in trustees tables by rebalancing column flex proportions and responsive breakpoints.

Enhancements:
- Refine trustee filter layout into paired, responsive controls that adapt between stacked, two-column, and multi-row arrangements.
- Standardize borders, text, and interaction colors across multiple components using shared SCSS color variables.
- Improve reusable CamsTable mixins and screen-specific table layouts for consistent stacked/flex behaviors at tablet and desktop breakpoints.
- Tidy various SCSS files for consistency, spacing, and selector formatting without changing behavior.

CI:
- Align ESLint testing configuration formatting and rules for Testing Library in UI test files.

Documentation:
- Reformat accessibility KNOWN_ISSUES documentation for better readability and updated status notes on existing issues.

Tests:
- Add unit tests covering trustee status filtering combinations, status label formatting, and combined chapter/status filter behavior.
- Extend district filter tests to cover the new Status combobox wiring and default selection.
- Adjust existing trustees list tests to account for status cell rendering changes and new inactive status labeling.

### CODE REVIEW

**Review by mattstankey — COMMENTED** [review]
Code review feedback on four items found during review.

File: user-interface/src/trustees/TrusteesList.tsx [hunk 1]
```diff
@@ -55,17 +61,22 @@ function filterTrustees(
...
   return trustees.filter((trustee) => {
+    if (statusFilter === 'active') {
+      if (
+        trustee.appointments.length > 0 &&
+        !trustee.appointments.some((appt) => appt.status === 'active')
+      )
+        return false;
+    } else if (statusFilter === 'inactive') {
+      if (!trustee.appointments.some((appt) => appt.status !== 'active')) return false;
+    }
```
Comment (mattstankey) [inline user-interface/src/trustees/TrusteesList.tsx:78]: **Bug: Empty-appointment trustees behave asymmetrically across filter states**

The `active` and `inactive` branches treat trustees with zero appointments differently:

```ts
if (statusFilter === 'active') {
  if (
    trustee.appointments.length > 0 &&  // ← zero-appointment trustees pass through
    !trustee.appointments.some(...)
  )
    return false;
} else if (statusFilter === 'inactive') {
  if (!trustee.appointments.some(...)) return false;
  // .some() on an empty array returns false → !false = true → filtered OUT
}
```

- **Active filter**: a trustee with no appointments is *included*.
- **Inactive filter**: a trustee with no appointments is *excluded*.

If empty-appointment trustees are invalid, both branches should exclude them. If they are valid, both should treat them consistently. Consider removing the `appointments.length > 0` guard from the active branch (or mirroring it in the inactive branch), and add a test covering this edge case.

File: user-interface/src/trustees/TrusteesList.tsx [hunk 2]
```diff
@@ -55,17 +61,22 @@ function filterTrustees(
   trustees: TrusteeListItem[],
   selectedDistricts: ComboOption[],
   selectedChapters: ComboOption[],
+  statusFilter: StatusFilterValue,
   districtDivisionEnabled: boolean = false,
```
Comment (mattstankey) [inline user-interface/src/trustees/TrusteesList.tsx:65]: **Design: `filterTrustees` is called 6+ times in this component**

`statusFilter` is now threaded through every call correctly, but having this many separate calls to `filterTrustees` is fragile — a future parameter addition will require finding all call sites again.

The existing `useMemo` infrastructure in this file is the right tool here. Computing `filteredTrustees` once and reusing it for display, pagination, and counts would simplify the component and align with how other filters are handled:

```ts
const filteredTrustees = useMemo(
  () => filterTrustees(trustees, selectedDistricts, selectedChapters, statusFilter, districtDivisionEnabled, divisionFilterMap),
  [trustees, selectedDistricts, selectedChapters, statusFilter, districtDivisionEnabled, divisionFilterMap],
);
```

File: user-interface/src/trustees/TrusteesList.test.tsx [hunk 3]
```diff
@@ -2958,4 +2970,204 @@ describe('TrusteesList Component', () => {
...
+      });
+
+      renderWithRouter(<TrusteesList />);
+
+      await waitFor(() => {
+        expect(screen.getByText('1 Trustee', { selector: 'p' })).toBeInTheDocument();
+      });
+
+      const toggleButton = screen.getByRole('button', { name: /filters/i });
+      await userEvent.setup().click(toggleButton);
```
Comment (mattstankey) [inline user-interface/src/trustees/TrusteesList.test.tsx:3033]: **Test pattern: `userEvent.setup()` should be called once per test and reused**

Each call to `userEvent.setup()` creates a new user-event instance with its own event queue. Calling it separately for each interaction can cause issues with pointer state, delays, and event ordering.

```ts
// Current (problematic)
await userEvent.setup().click(toggleButton);
// ...
await userEvent.setup().click(statusCombobox);

// Correct
const user = userEvent.setup();
await user.click(toggleButton);
// ...
await user.click(statusCombobox);
```

The `should apply status and chapter filters together` test already does this correctly — the other three new tests in this suite should follow the same pattern.

File: user-interface/src/trustees/filters/TrusteeDistrictFilterView.tsx [hunk 4]
```diff
@@ -2,7 +2,22 @@ import './TrusteeDistrictFilter.scss';
...
+import { StatusFilterValue, TrusteeDistrictFilterViewProps } from './trusteeDistrictFilter.types';
+
+const STATUS_OPTIONS: ComboOption[] = [
+  { value: 'all', label: 'All' },
+  { value: 'active', label: 'Active' },
+  {
+    value: 'inactive',
+    label:
+      'Inactive (Inactive, Involuntarily Suspended, Voluntarily Suspended, Resigned, Terminated, Deceased)',
+  },
```
Comment (mattstankey) [inline user-interface/src/trustees/filters/TrusteeDistrictFilterView.tsx:14]: **UX: The Inactive dropdown label is very long and will likely truncate**

```ts
label: 'Inactive (Inactive, Involuntarily Suspended, Voluntarily Suspended, Resigned, Terminated, Deceased)',
```

This string is ~82 characters and will overflow or truncate in most combobox dropdown widths. Consider shortening the label to just `"Inactive"` (consistent with the brevity of `"All"` and `"Active"`) and surfacing the sub-status breakdown via tooltip, help text, or the status column itself in the list.
